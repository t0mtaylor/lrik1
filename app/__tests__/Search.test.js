
import React from 'react';
import { shallow } from 'enzyme';


import Navbar from '../src/Search.js';


const stub_hotels = [
  {
    'name': 'hotelnine',
    'starRating': 1,
    'facilities': ['car park', 'pool']
  },
  {
    'name': 'hotelseven',
    'starRating': 3,
    'facilities': ['wifi', 'late bar']
  }
];


describe('Search tests', () => {
	
	beforeEach(() => {
		fetch.resetMocks()
		fetch.mockResponseOnce(JSON.stringify(stub_hotels))
	})
	
	it('renders without crashing', () => {

		const 	navbar = shallow(<Navbar />),
				header = <h2>Search Results</h2>;

		expect(navbar.contains(header)).toEqual(true);
		
		expect(navbar).toMatchSnapshot();
	});
	
	it('renders with data', () => {

		shallow(<Navbar />);	
				
		expect(fetch.mock.calls.length).toEqual(1);
		expect(fetch.mock.calls[0][0]).toEqual('./data/data.json');
	});
	
});
