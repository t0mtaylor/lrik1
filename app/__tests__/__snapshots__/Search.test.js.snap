// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <img
            alt="LR.com"
            src="logo-laterooms.com.png"
            width="200"
          />
          <h1>
            lrik1 - hotel results
          </h1>
        </header>,
        <div
          className="search"
        >
          <h2>
            Search Results
          </h2>
          <div
            className="content is-loading"
          >
            <div
              className="panel-group"
            />
            <div
              className="loader"
            >
              <div
                className="icon"
              />
            </div>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="LR.com"
              src="logo-laterooms.com.png"
              width="200"
            />,
            <h1>
              lrik1 - hotel results
            </h1>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "LR.com",
              "src": "logo-laterooms.com.png",
              "width": "200",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "lrik1 - hotel results",
            },
            "ref": null,
            "rendered": "lrik1 - hotel results",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Search Results
            </h2>,
            <div
              className="content is-loading"
            >
              <div
                className="panel-group"
              />
              <div
                className="loader"
              >
                <div
                  className="icon"
                />
              </div>
            </div>,
          ],
          "className": "search",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search Results",
            },
            "ref": null,
            "rendered": "Search Results",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="panel-group"
                />,
                <div
                  className="loader"
                >
                  <div
                    className="icon"
                  />
                </div>,
              ],
              "className": "content is-loading",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                  "className": "panel-group",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="icon"
                  />,
                  "className": "loader",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <img
              alt="LR.com"
              src="logo-laterooms.com.png"
              width="200"
            />
            <h1>
              lrik1 - hotel results
            </h1>
          </header>,
          <div
            className="search"
          >
            <h2>
              Search Results
            </h2>
            <div
              className="content is-loading"
            >
              <div
                className="panel-group"
              />
              <div
                className="loader"
              >
                <div
                  className="icon"
                />
              </div>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="LR.com"
                src="logo-laterooms.com.png"
                width="200"
              />,
              <h1>
                lrik1 - hotel results
              </h1>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "LR.com",
                "src": "logo-laterooms.com.png",
                "width": "200",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "lrik1 - hotel results",
              },
              "ref": null,
              "rendered": "lrik1 - hotel results",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Search Results
              </h2>,
              <div
                className="content is-loading"
              >
                <div
                  className="panel-group"
                />
                <div
                  className="loader"
                >
                  <div
                    className="icon"
                  />
                </div>
              </div>,
            ],
            "className": "search",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search Results",
              },
              "ref": null,
              "rendered": "Search Results",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="panel-group"
                  />,
                  <div
                    className="loader"
                  >
                    <div
                      className="icon"
                    />
                  </div>,
                ],
                "className": "content is-loading",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": null,
                    "className": "panel-group",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="icon"
                    />,
                    "className": "loader",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "icon",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
